timestamp/request.go

```diff
@@ -37,36 +38,38 @@ type Request struct {
 	Extensions     []pkix.Extension      `asn1:"optional,tag:0"`
 }
 
-// NewRequest creates a request based on the given digest.
-func NewRequest(contentDigest digest.Digest) (*Request, error) {
-	hashAlgorithm, err := getOIDFromDigestAlgorithm(contentDigest.Algorithm())
+// NewRequest creates a request based on the given digest and hash algorithm.
+func NewRequest(digest []byte, alg crypto.Hash) (*Request, error) {
```

Comment:
Is there a reason we need both NewRequest/NewRequestWithData , Verify/VerifyWithData ?